aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/SRPCognitoActions.kt

```diff
@@ -64,7 +65,7 @@ internal object SRPCognitoActions : SRPActions {
                 secretHash?.let { authParams[KEY_SECRET_HASH] = it }
 
                 val encodedContextData = getUserContextData(event.username)
-                val deviceMetadata = getDeviceMetadata(event.username)
+                var deviceMetadata = getDeviceMetadata(event.username)
```

Comment:
Looks like deviceMetadata is only used in the password verifier block. Just create deviceMetadata in there and move the next line to assign KEY_DEVICE_KEY down in the when block.

Reply:
No it is used in line 68(now 69) as well : See https://github.com/aws-amplify/amplify-android/pull/2614/files#diff-f40747d07791c3bcc88babaae31fcee22acf2c435db334d77fda44eb30293ee7R69